consensus/core/src/leader_schedule.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use rand::{prelude::SliceRandom, rngs::StdRng, SeedableRng};
+use std::sync::Arc;
+
+use consensus_config::{AuthorityIndex, Committee};
+
+/// The LeaderSchedule is responsible for producing the leader schedule across
+/// an epoch. For now it is a simple wrapper around committee to provide a leader
+/// for a round deterministically.
+// TODO: complete full leader schedule changes
+#[derive(Clone)]
+pub struct LeaderSchedule {
+    committee: Arc<Committee>,
```

Comment:
Use Arc<Context> instead, so it will be easier to use metrics and protocol config later.